src/rest_offsets.lua

```diff
@@ -87,23 +86,40 @@ function make_the_change()
     if finenv.RetainLuaState ~= nil then
         finenv.RetainLuaState = true
     end
+    local current_staff = nil
+    local staff_spec = finale.FCCurrentStaffSpec()
+    local rest_type = { "OtherRestPosition", "HalfRestPosition", "WholeRestPosition", "DoubleWholeRestPosition" }
+
     for entry in eachentrysaved(finenv.Region(), config.layer) do
          if entry:IsRest() then
             if config.offset == 0 then
                 entry:SetFloatingRest(true)
             else
+                if current_staff ~= entry.staff then -- need a new staff spec
+                    current_staff = entry.staff
+                    staff_spec:LoadForEntry(entry)
+                end
+                local duration = entry.Duration
+                if duration % 3 == 0 then -- it's dotted
+                    duration = duration * 2 / 3 -- get the un-dotted version
+                end
+                local power_count = 1
+                while duration > 1024 and power_count <= #rest_type do
+                    duration = duration / 2
+                    power_count = power_count + 1
+                end
+                local rest_type_offset = staff_spec[rest_type[power_count]] + 4 -- adjusted for middle line
                 entry:MakeMovableRest()
```

Comment:
Generally speaking, a good rule of thumb is to avoid hard-coding numbers other than 0. It is impossible to make that a hard-and-fast rule, but you can get close. Most of these hard-coded numbers could have been avoided with finale constants. It would be much clearer to write:

```lua
if duration >= finale.DOUBLE_WHOLE then
   rest_prop = "DoubleWholeRestPosition"
elseif duration >= finale.WHOLE_NOTE then
   rest_prop = "WholeRestPosition"
elseif duration >= finale.HALF_NOTE then
   rest_prop = "HalfRestPosition"
else
   rest_prop = "OtherRestPosition",
end

local rest_type_offset = staff_spec[rest_prop] + 4 -- WHY 4??!! (see below)
```

That's it. No need to check dotted (unless you need it for something else). And no hard-coded numbers.

The one that really bugs me is line 111. What is that magic 4? The `staff_spec` for  standard staves has a rest position of -4 for all rests. So in the most common case  `rest_type_offset` will end up zero. Why are you adding 4? Have you tested this code?

Reply:
https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants

Reply:
Will amend with chained elseif's. Got you both on the un-magic of fixed constants. I decided (arbitrarily) to call the middle staff position zero (input by user) in this script, since that's where rests "appear" to sit. `rest_type_offset` is subtracted from the final offset so the plus four moves it down the staff. Yes - its was tested and works with all basic staves and rest durations. Should I leave zero to represent the top staff line?